src/binary-search-tree.js

```diff
@@ -5,6 +5,49 @@ class BinarySearchTree {
     this.left = null;
     this.right = null;
   }
+  insert(value) {
+    const node = new BinarySearchTree(value);
+    function recurse(bst) {
```

Comment:
This recursive function never returns, so it will blow the stack whenever it is executed. You need to ensure that this function's base case will eventually be reached, and then inside that base case, that the recursive function returns.